components/sup/src/ctl_gateway/server.rs

```diff
@@ -285,6 +285,27 @@ impl Future for SrvHandler {
                                             },
                                         )
                                     }
+                                    "SvcSetCfg" => {
+                                        let m = msg.parse::<protocols::ctl::SvcSetCfg>().unwrap();
```

Comment:
I was trusting the client to send the payload they say they are because they've already passed validation. I probably don't want to crash this thread though since I don't have a recovery mechanism. I'll just handle the decoding errors - good catch!